src/fetch/transfer_file.py

```diff
@@ -141,7 +152,19 @@ def erc20(
     @property
     def amount(self) -> float:
         """Returns transfer amount in units"""
-        return self.amount_wei / 10**18
+        return self.amount_wei / 10 ** self.token_decimals()
+
+    def token_decimals(self) -> int:
+        """Fetches token decimals from EVM when not already known."""
+        if self.token_type == TokenType.NATIVE:
+            return 18
+        try:
+            return cached_token_decimals(self.token_address)
+        except web3.exceptions.BadFunctionCallOutput as err:
+            logging.error(
+                f"failed to fetch token decimals for {self.token_address}. returning 18"
+            )
+            return 18
```

Comment:
This is a cheeky hack for testing dummy accounts. I would encourage alternate suggestions here.

Reply:
One thing I could do is include an Optional field decimals on the object constructor which defaults to None and gets set whenever it is needed via this method. However, it would allow us to pass decimals in with the constructor during unit tests. Then we wouldn't have to fall back on this hacky default.

Reply:
Resolved.